middle_end/flambda2/from_lambda/lambda_to_flambda_primitives_helpers.ml

```diff
@@ -490,3 +492,29 @@ and bind_rec_primitive acc exn_cont ~register_const0 (prim : simple_or_prim)
         (acc, body) (List.rev vars') (List.rev nameds)
     in
     bind_recs acc exn_cont ~register_const0 p dbg cont
+
+let convert_block_access_field_kind_from_value_kind
+    ({ raw_kind; nullable = _ } : L.value_kind) : P.Block_access_field_kind.t =
+  match raw_kind with
+  | Pintval -> Immediate
+  | Pvariant { consts = _; non_consts } -> (
+    match non_consts with [] -> Immediate | _ :: _ -> Any_value)
+  | Pgenval | Pboxedfloatval _ | Pboxedintval _ | Parrayval _
+  | Pboxedvectorval _ ->
+    Any_value
+
+let block_access_for_element ~for_block_load
+    (elt : 'a Mixed_block_shape.Singleton_mixed_block_element.t) :
+    P.Mixed_block_access_field_kind.t =
+  match elt with
+  | Value value_kind ->
+    Value_prefix
+      (if for_block_load
```

Comment:
It's not obvious to me what the `for_block_load` parameter is doing for us.  Why do we want to skip computing a value kind in that case?

My hypothesis is that this is an optimization, since you know you haven't computed accurate value kinds in some cases. But I don't think this optimization is buying us enough to be worth doing - it's just turning a pattern match into an `if`, and we may find we want to compute better value kinds in lambda later.

(If we do want to keep this parameter, I think its name should change - currently we pass `~for_black_load:false` in the compilation of `Pmixedfield`, but that is also a load.